homeassistant/components/alarmdecoder/__init__.py

```diff
@@ -110,6 +112,9 @@
                 vol.Optional(
                     CONF_CODE_ARM_REQUIRED, default=DEFAULT_CODE_ARM_REQUIRED
                 ): cv.boolean,
+                vol.Optional(
+                    CONF_ALT_NIGHT_MODE, default=DEFAULT_ALT_NIGHT_MODE
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

I suggest removing this option for now so we can go ahead with this PR. Then make a separate PR where config entries and config flow is introduced.

Reply:
@MartinHjelmare any chance I can get an exception here? I introduced this functionality previously in https://github.com/home-assistant/core/pull/35761 and was told that it needed to be moved to a separate library, which I've now done. I'm eager to get this merged because I've heard from people who were unhappy about the changes I made in https://github.com/home-assistant/core/pull/32292 and this will give them a way to revert those changes. It's starting to seem like the bars are getting raised here every time I open a PR. This is also the second time I've had to open up my long documentation PR here https://github.com/home-assistant/home-assistant.io/pull/13725. I'm not looking forward to revising it now and opening it up a 3rd time for the same functionality.

My next task is to bring multiple partition support to AlarmDecoder which will require more config changes, so I'd be happy to introduce the config flow then. But I'm really hoping you can make an exception for this code.

Reply:
Sorry, we're following our design guidelines and architecture decisions. When the project develops our standards evolve and changes to integrations need to follow the new standards.